### PR TITLE
switch from `tap` to `mocha`

### PR DESCRIPTION
**Pull request by juliangruber**
We need the extra stability that `mocha` has to offer

### CODE REVIEW

**Comment by juliangruber** [comment]
ping @bajtos

File: test/test.js [hunk 1]
```diff
@@ -7,79 +6,86 @@ import fs from 'node:fs/promises'
...
-  await t.test('require address', async t => {
+describe('FIL_WALLET_ADDRESS', () => {
+  it('fails without address', async () => {
     try {
       await execa(station)
     } catch (err) {
       return
     }
-    t.fail('should have thrown')
+    throw new Error('should have thrown')
```
Comment (bajtos) [inline test/test.js:24]: Slightly unrelated to this PR, but have you considered using [`assert.rejects`](https://nodejs.org/dist/latest-v18.x/docs/api/assert.html#assertrejectsasyncfn-error-message)?

```js
assert.rejects(execa(station))
```

**Review by bajtos — APPROVED** [review]
Nice! 💪🏻

File: test/test.js (hunk 1 above)
Comment (juliangruber) [reply test/test.js]: That is awesome, thank you so much for the tip!!

File: test/test.js [hunk 2]
```diff
@@ -367,15 +362,17 @@ describe('Lockfile', () => {
     const ps = execa(station, { env: { ROOT_DIR, FIL_WALLET_ADDRESS } })
     await once(ps.stdout, 'data')
     try {
-      await execa(station, { env: { ROOT_DIR, FIL_WALLET_ADDRESS } })
-    } catch (err) {
-      assert.strictEqual(err.exitCode, 1)
-      assert.match(err.stderr, /is already running/)
-      return
+      await assert.rejects(
```
[the remark this answers is not in the document]
Comment (juliangruber) [reply test/test.js:365]: I think it doesn't, I tried exactly that first

**Pull request merged**